test/test_module_init.py

```diff
@@ -0,0 +1,330 @@
+
+                # == Verify kwargs logic. ==
+                if module_creates_params_or_buffers and module_cls not in MODULES_WITHOUT_KWARGS_SUPPORT:
+                    args, kwargs = get_example_args(module_cls, constructor_arg_db, device=device)
+
+                    # if module_cls in LAZY_MODULES:
+                    #     # Ensure device is passed to all UninitializedParameters and UninitializedBuffers.
+                    #     # TODO: Check dtype!
+                    #     uninit_param_new = mock_wrapper(torch.nn.UninitializedParameter.__new__)
+                    #     with patch.object(torch.nn.UninitializedParameter, '__new__', uninit_param_new):
+                    #         uninit_buffer_new = mock_wrapper(torch.nn.UninitializedBuffer.__new__)
+                    #         with patch.object(torch.nn.UninitializedBuffer, '__new__', uninit_buffer_new):
+                    #             m = module_cls(*args, **kwargs)
+                    #             uninit_param_new.mock.assert_has_calls(
+                    #                 [unittest.mock.call(device=device) for _ in uninit_param_new.mock.mock_calls])
+                    #             uninit_buffer_new.mock.assert_has_calls(
+                    #                 [unittest.mock.call(device=device) for _ in uninit_buffer_new.mock.mock_calls])
+                    if device == 'meta':
+                        # TODO: Once meta tensors support in-place operations, don't expect an error.
+                        with test_cls.assertRaisesRegex(NotImplementedError,
+                                                        'is only available for these backends'):
+                            m = module_cls(*args, **kwargs)
+                    else:
+                        # Check device placement for parameters and buffers.
+                        m = module_cls(*args, **kwargs)
+                        for name, param in m.named_parameters():
+                            test_cls.assertEqual(param.device.type, device,
+                                f'Parameter {name} is on {param.device.type} instead of the expected device {device}')
+                        for name, buffer in m.named_buffers():
+                            test_cls.assertEqual(buffer.device.type, device,
+                                f'Buffer {name} is on {param.device.type} instead of the expected device {device}')
+                        # TODO: Check dtype!
+
+                # == Verify passing a nonexistent arg still errors out if it did before. ==
+                if module_cls not in MODULES_WITH_PREVIOUS_KWARGS:
+                    with test_cls.assertRaises(TypeError):
+                        m = module_cls(*args, **kwargs, nonexistent_arg='foo')
+
+            test_name = f'test_{namespace_basename}_{module_name}'
+            setattr(TestModuleInit, test_name, run_test)
+
+    instantiate_device_type_tests(test_cls, globals())
```

Comment:
I would put this call on the current line 327 for readability and consistency with other test files